libraries/AP_Mount/AP_Mount_Backend.cpp

```diff
@@ -26,27 +26,40 @@ void AP_Mount_Backend::set_roi_target(const struct Location &target_loc)
     _frontend.set_mode(_instance, MAV_MOUNT_MODE_GPS_POINT);
 }
 
-// configure_msg - process MOUNT_CONFIGURE messages received from GCS
+// configure_msg - process MOUNT_CONFIGURE messages received from GCS.  deprecated.
 void AP_Mount_Backend::configure_msg(mavlink_message_t* msg)
 {
     __mavlink_mount_configure_t packet;
     mavlink_msg_mount_configure_decode(msg, &packet);
 
-    // set mode
-    _frontend.set_mode(_instance,(enum MAV_MOUNT_MODE)packet.mount_mode);
+    configure((MAV_MOUNT_MODE)packet.mount_mode, packet.stab_roll, packet.stab_pitch, packet.stab_yaw);
+}
 
+void AP_Mount_Backend::configure(MAV_MOUNT_MODE mount_mode, uint8_t stab_roll, uint8_t stab_tilt, uint8_t stab_pan)
+{
+    // set mode
+    _frontend.set_mode(_instance, mount_mode);
     // set which axis are stabilized
-    _state._stab_roll = packet.stab_roll;
-    _state._stab_tilt = packet.stab_pitch;
-    _state._stab_pan = packet.stab_yaw;
+    _state._stab_roll = stab_roll;
+    _state._stab_tilt = stab_tilt;
+    _state._stab_pan = stab_pan;
 }
 
-// control_msg - process MOUNT_CONTROL messages received from GCS
+// control_msg - process MOUNT_CONTROL messages received from GCS. deprecated.
 void AP_Mount_Backend::control_msg(mavlink_message_t *msg)
 {
     __mavlink_mount_control_t packet;
     mavlink_msg_mount_control_decode(msg, &packet);
 
+    control((int32_t) packet.input_a, (int32_t) packet.input_b, (int32_t) packet.input_c, 0);
+}
+
+void AP_Mount_Backend::control(int32_t pitch_or_lat, int32_t roll_or_lon, int32_t yaw_or_alt, uint8_t mount_mode)
+{
+    if(mount_mode > 0) {
```

Comment:
Nevermind, I just saw you change that leaves the mode the same if the old message is used, but requires the mode to be set with the new message.  That seems like a good solution.